### ISSUE TITLE
transition_reveal() messes up geom_path() in version 1.0.8

### ISSUE THREAD

**Issue by yutannihilation** [thomasp85/gganimate#473]
Sorry for filing an issue with not-so-minimal reprex. It seems some behavior around `transition_reveal()` changed unexpectedly on version 1.0.8. As this is about `transition_reveal()`, this might be the same problem as  #468, but I'm filing a new one in case this is a different problem.

## Expected (generated with gganimate version 1.0.7)

![](https://raw.githubusercontent.com/yutannihilation/string2path/main/man/figures/README-example-1.gif)

## Actual

``` r
library(string2path)
library(ggplot2)
library(gganimate)

d <- string2path("蹴", "Noto Sans JP")
d <- tibble::rowid_to_column(d)

ggplot(d) +
  geom_path(aes(x, y, group = path_id),
            linewidth = 2, colour = "purple2", lineend = "round") +
  theme_minimal() +
  coord_equal() +
  transition_reveal(rowid)
#> `geom_path()`: Each group consists of only one observation.
#> ℹ Do you need to adjust the group aesthetic?
#> `geom_path()`: Each group consists of only one observation.
#> ℹ Do you need to adjust the group aesthetic?
```

![](https://i.imgur.com/EeKNBcW.gif)

<sup>Created on 2022-12-17 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by yutannihilation** [comment]
It seems `transition_reveal()` skips some data points for some reason?

``` r
library(string2path)
library(ggplot2)
library(gganimate)

d <- string2path("&", "Arial", tolerance = 1e-4)
d <- tibble::rowid_to_column(d)
d2 <- dplyr::select(d, !rowid)

red <- alpha("red", 0.2)

ggplot(mapping = aes(x, y)) +
  geom_path(data = d2, aes(group = path_id), colour = red) +
  geom_point(data = d2, colour = red) +
  geom_path(data = d, aes(group = path_id)) +
  geom_point(data = d) +
  theme_minimal() +
  coord_equal() +
  transition_reveal(rowid)
#> `geom_path()`: Each group consists of only one observation.
#> ℹ Do you need to adjust the group aesthetic?
#> `geom_path()`: Each group consists of only one observation.
#> ℹ Do you need to adjust the group aesthetic?
```

![](https://i.imgur.com/0dcqjgv.gif)

<sup>Created on 2022-12-17 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by mccarthy-m-g** [comment]
See also the reprex in #445 which was broken in v1.0.8

**Comment by ddeannd** [comment]
Figured I would add to this for another test case. I wish I checked here before fumbling around for a few hours trying to figure out what was wrong. A simple line chart with geom_line() also suffers from this bug, but only on the final frame from what I can tell. Every line but one will revert to it's previous value, no matter how long the x axis is. Here is an example: 

![test](https://user-images.githubusercontent.com/121740869/210160300-6068ffa0-8a2e-4979-8355-b92abf916096.gif)

I made a post on stackoverflow regarding this, as well. https://stackoverflow.com/questions/74971922/gganimate-transition-reveal-with-geom-line-breaking-on-the-final-frame
